pkg/acpi/ibft.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package acpi
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+)
+
+//The acpiIBFT is another braind-dead design.
+//Basic layout is like this:
+//	Name		size		offset		description
+//	Header		48		0		Primary Header
+//	Control		variable	48		Extended Header
+//	Initiator	variable			Initiator
+//	NIC		variable			NIC
+//	Target		variable			Target
+//	Heap		variable			Offsets point here.
+
+const (
+	reserved uint8 = iota
+	ibftControl
+	ibftInitiator
+	ibftNIC
+	ibftTarget
+	ibftExtensions
+	ibftControlLen   = 18
+	ibftInitiatorLen = 74
+	ibftNICLen       = 102
+	ibftTargetLen    = 54
+	ibftVersion      = 1 // in all cases since 2009
+)
+
+// We created a bunch of structs here so you don't have to go read the Big Bad Book of ACPI.
+// It turned out to be easier to use the structs we defined further below.
+// This was generated by running scripts across the pdf.
+
+// acpiIBFTHeader defines the acpiIBFTHeader
+type acpiIBFTHeader struct {
+	Signature  [4]byte  `offset:"0" desc:"iBFT Signature for the iSCSI Boot Firmware Table"`
+	Length     uint32   `offset:"4" desc:"Length in bytes of the entire IBFT, including the signature"`
+	Revision   uint8    `offset:"8" desc:"Revision = 1"`
+	Checksum   uint8    `offset:"9" desc:"Entire table must sum to zero"`
+	OEMID      [6]byte  `offset:"10" desc:"OEM ID. All unused trailing bytes must be zero. [acpi-OEMID]"`
+	OEMTableID [8]byte  `offset:"16" desc:"For the iBFT the Table ID is the Manufacturer‟s Model ID. All unused trailing bytes must be zero."`
+	Reserved   [24]byte `offset:"24" desc:"Reserved"`
+}
+
+var rawIBTFHeader = "IBFT\x00\x08\x00\x001\x00ACPIXXACPISUCK1234567890abcdef01234567"
```

Comment:
```suggestion
var rawIBTFHeader = "IBFT\x00\x08\x00\x001\x00ACPIXXACPISUCKS123456789abcdef01234567"
```

Reply:
no, the ACPISUCK exactly matches a field in ACPI. The length has to be 8.